src/php/Reader.php

```diff
@@ -88,7 +79,7 @@ private function readWhitespace(Generator $tokenStream): void
     }
 
     /**
-     * @return Phel|null|scalar
+     * @return Phel|null|string|scalar
```

Comment:
`string` is unnecessary, because it is a `scalar` :)
Or do I miss something?

See https://www.php.net/manual/en/function.is-scalar.php

Reply:
You're right. This is a separate "problem".
PHPDoc doesn't recognize scalar as type for `string|bool|int|float`, in fact:
* some people already tried to create some special `csfixer` for this https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3341 
* and even adding `scalar` Pseudo-type to the php-src: https://github.com/php/php-src/pull/2987

both of them declined.

For this reason, I'm a bit reluctant to use `Phel|null|scalar` because it's mixing too many types at once. But, as I said, that's another epic. I actually have an idea about how could we improve that, but not in this PR ;)

I will revert this line leaving how it was before.